botocore/signers.py

```diff
@@ -11,6 +11,15 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 import datetime
+import logging
+import json
+import base64
+try:
```

Comment:
Lets move this to ``compat.py``. That is where we put all of these type of conversions and ensures we never have to repeat the logic. All you have to do is import it from ``botocore.compat``